### PR TITLE
remove not used plugin heartbeat.lua

### PR DESCRIPTION
**Pull request by mtx2d**
Fix #1176 

### Pre-submission checklist:

*Please answer each of these after submitting your pull request:*

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
#1776 Says that heartbeat is no longer needed, we can remove this. 
* [x] Have you added corresponding test cases?
Checked that there is not test cases related to heartbeat.
Removing this file does not break existing test suit.
* [x] Have you modified the corresponding document?
Link to document including [`heartbeart`](https://github.com/apache/incubator-apisix/search?l=Markdown&q=heartbeat). 
There are three instances, two of them in the change log, and last one discussing in general service discovery. 
No need to modify any one of them.
* [x] Is this PR backward compatible?
Not sure how to check this.

### CODE REVIEW

**Comment by membphis** [comment]
@mtx2d please take a look at the output of CI, eg: https://github.com/apache/incubator-apisix/pull/1845/checks?check_run_id=871655239#step:6:272

**Comment by mtx2d** [comment]
> @mtx2d please take a look at the output of CI, eg: https://github.com/apache/incubator-apisix/pull/1845/checks?check_run_id=871655239#step:6:272

Cannot run the test locally, so running this with CI. 
In the meantime, could you please say more on why we no longer need heartbeat?
I can read the test failure and all, but design wise understanding the motivation will help me to create the correct PR.

File: conf/config.yaml [hunk 1]
```diff
@@ -16,7 +16,7 @@
 #
 apisix:
   node_listen: 9080              # APISIX listening port
-  enable_heartbeat: true
+  enable_heartbeat: false
```
Comment (membphis) [inline conf/config.yaml:19]: we need to remove this line.

some code we need to remove too: 

https://github.com/apache/incubator-apisix/search?p=2&q=enable_heartbeat&unscoped_q=enable_heartbeat

Comment (mtx2d) [reply conf/config.yaml:19]: Thanks for taking time to review, will do.

File: t/lib/server.lua [hunk 2]
```diff
@@ -112,10 +112,6 @@ function _M.mock_skywalking_v2_instance_register()
     ngx.say(json_encode(ret))
 end
 
-function _M.mock_skywalking_v2_instance_heartbeat()
-    ngx.say('skywalking heartbeat ok')
-end
-
```
Comment (moonming) [inline t/lib/server.lua:118]: these codes should not remove.

File: Makefile [hunk 3]
```diff
@@ -176,7 +176,7 @@ install: default
 
 ### test:             Run the test case
 test:
-	prove -I../test-nginx/lib -I./ -r -s t/
+	TEST_NGINX_BINARY=`which openresty` prove -I../test-nginx/lib -I../test-nginx/inc -I./ -r -s t/
```
Comment (moonming) [inline Makefile:179]: why change this line? which is non-related with heartbeat plugin.

File: t/lib/server.lua (hunk 2 above)
Comment (membphis) [reply t/lib/server.lua:118]: Need to revert these codes, then the test case will run successfully.

```
Test Summary Report
-------------------
t/plugin/skywalking.t                (Wstat: 256 Tests: 30 Failed: 1)
  Failed test:  10
  Non-zero exit status: 1
Files=120, Tests=4005, 1041 wallclock secs ( 1.02 usr  0.16 sys + 152.60 cusr 29.88 csys = 183.66 CPU)
Result: FAIL
##[error]Process completed with exit code 1.
```

Comment (mtx2d) [reply t/lib/server.lua:118]: reverted in latest commit.

File: Makefile (hunk 3 above)
Comment (mtx2d) [reply Makefile:179]: Reverted in new commit.

**Comment by mtx2d** [comment]
Trying to figure out how to trigger CI.

**Comment by mtx2d** [comment]
@moonming @membphis , is it possible to force trigger a CI? 
I made new commit to my local repo mtx2d/incubato-apisix, but the CI still looks at the old states.(my master should be 11 commits ahead but the CI still believes I am only 5 commits ahead). 


How can I force trigger a CI?

**Comment by mtx2d** [comment]
> @moonming @membphis , is it possible to force trigger a CI?
> I made new commit to my local repo mtx2d/incubato-apisix, but the CI still looks at the old states.(my master should be 11 commits ahead but the CI still believes I am only 5 commits ahead).
> 
> How can I force trigger a CI?

Seems like the CI picks it up now.

**Comment by mtx2d** [comment]
The CI fails due to CI server network issue: https://github.com/apache/incubator-apisix/pull/1845/checks?check_run_id=874200614#step:3:68

Will make a small change and trigger another CI run.

**Comment by mtx2d** [comment]
According to this [doc](https://github.com/travis-ci/travis-ci/issues/576#issuecomment-65317915), we can close and reopen, which will trigger a new CI run without making changes.

**Comment by mtx2d** [comment]
Hi @membphis, 
Do we think this is PR good to move forward :D?

**Comment by membphis** [comment]
@mtx2d many thx, merged already

**Pull request merged**